bluesky/tiffexporter.py

```diff
@@ -0,0 +1,263 @@
+        Header.  Use ``naming(h)`` to obtain a list of output names.
+        To adjust naming scheme, modifying the `naming.template` and
+        `naming.prefix` attributes.
+    use_mtime : bool
+        When True (default), write output files with modification time set
+        according to the exported Event.  Also use modification time to
+        identify existing outputs.  When False, write files with current
+        modification time and only check if target paths exist.
+        Default is `True`.
+    mtime_window : float
+        Precision in file modification times when searching for existing
+        output files.   Ignored when `use_mtime` is `False`.
+
+    Parameters
+    ----------
+    fetch : str or callable object, optional
+        Set function that will generate 2D arrays from databroker Header.
+        When string, obtain arrays using ``get_images(h, NAME)``.
+        Use the `default_fetch` value when not specified.
+    template : str, optional
+        Set `naming.template`, a filename template where curly brackets
+        are filled with entries from databroker Header.  See the DataNaming
+        class for more details.
+    prefix : str, optional
+        Set `naming.prefix` a constant output directory that is prefixed
+        to the generated filenames.
+        Default is ''.
+
+    See Also
+    --------
+    bluesky.datanaming.DataNaming : file name generation from a header.
+    """
+
+    # class attributes and defaults
+
+    default_fetch = 'pe1_image_lightfield'
+    use_mtime = True
+    mtime_window = 0.05
+
+
+    def __init__(self, fetch=None, template=None, prefix=None):
+        self.fetch = self.default_fetch if fetch is None else fetch
+        self.naming = DataNaming(template, prefix)
+        return
+
+
+    def __repr__(self):
+        nm = type(self).__name__
+        rv = ("<{0} fetch:{1.fetch} naming:{1.naming} "
+               "use_mtime:{1.use_mtime}>").format(nm, self)
+        return rv
+
+
+    def __call__(self, h, select=None, dryrun=False, overwrite=False):
+        """Export sequentially-numbered TIFF files from databroker header.
+
+        Parameters
+        ----------
+        h : Header
+            a header from the databroker
+        select : integer indices or mask array or slice, optional
+            Export image arrays only at corresponding indices, for example,
+            ``[4, 5]`` or ``numpy.s_[-3:]``.  Export all images by default.
+        dryrun : bool, optional
+            When True, display what would be done without taking any
+            action.
+        overwrite : bool, optional
+            Replace existing tiff files when True.
+
+        Returns
+        -------
+        list
+            Exported output files.
+        """
+        import tifffile
+        from databroker import get_events
+        if dryrun:
+            dryordo = lambda msg, f, *args, **kwargs:  print('[dryrun]', msg)
```

Comment:
Assigning lambdas to variables is discouraged.  Can you do this with two private module level functions?